pkg/certificates/certs.go

```diff
@@ -35,9 +35,8 @@ import (
 var randReader = rand.Reader
 var serialNumberLimit = new(big.Int).Lsh(big.NewInt(1), 128)
 
-// Copy-pasted from https://github.com/knative/pkg/blob/975a1cf9e4470b26ce54d9cc628dbd50716b6b95/webhook/certificates/resources/certs.go
+// Create template common to all certificates
```

Comment:
Are we planning to provide any abstractions for admins to inject their own short-term ca for all parties they are interested of? What is the api & docs for integrating with other cert management systems, any plan for that?

Reply:
This PR is focusing elsewhere. You bring up an important point, but I think we need to better define the requirements  - for example, (1) do we want to stop `control-protocol` and let some other extension create the certificates? (2) do we want `control protocol` to work but we want to force it to use a specific CA? (3) any other requirements?

Let's move this to a document focusing on the requirements and then design it in.

Reply:
I think supporting other CAs is a good discussion (probably related to extending the Certificate abstraction in Knative networking).  Let's hold that for now until we have a plan written up, though.

I think this is a step forward, even if we end up replacing some part of this with an `internal` Certificate provider.